Wikipedia/Code/NotificationsCenterCommonViewModel+ActionExtensions.swift

```diff
@@ -168,11 +168,22 @@ extension NotificationsCenterCommonViewModel {
             
         }
     }
-
+    private func getActionType(namespace: PageNamespace) -> RemoteNotificationActionType {
+        if notification.type == .userTalkPageMessage {
+            return .userTalk
+        } else if namespace == .userTalk {
+            return .senderPage
+        } else if namespace == .talk {
+            return .articleTalk
+        } else {
+            return .article
+        }
+    }
```

Comment:
Yeah, this is the trickiest action type since that `titleAction` method could create several different types of actions. I think let's purely only look at the namespace here to simplify, and test heavily. If we're creating an action URL with the namespace "User_talk:", then that should cover your first conditional, no need to look at the notification type.

```suggestion
    private func getActionType(namespace: PageNamespace) -> RemoteNotificationActionType? {
       if namespace == .userTalk {
            return .userTalk
        } else if namespace == .talk {
            return .articleTalk
        } else if namespace == .user {
            return .senderPage
        } else if namespace == .main {
            return .article
        } else {
            return nil
        }
    }
```